core/src/main/scala/org/apache/spark/rdd/ParallelCollectionRDD.scala

```diff
@@ -127,18 +127,12 @@ private object ParallelCollectionRDD {
       })
     }
     seq match {
-      case r: Range.Inclusive => {
-        val sign = if (r.step < 0) {
-          -1
-        } else {
-          1
-        }
-        slice(new Range(
-          r.start, r.end + sign, r.step).asInstanceOf[Seq[T]], numSlices)
-      }
       case r: Range => {
-        positions(r.length, numSlices).map({
-          case (start, end) =>
+        val sign = r.isInclusive && (r.end == Int.MaxValue || r.end == Int.MinValue)
```

Comment:
Maybe I'm missing it but why is `IntMinValue` a special case here? Also the `({` on the next line is redundant. Just one is needed.